src/System.Private.CoreLib/shared/System/IO/MemoryStream.cs

```diff
@@ -226,18 +226,21 @@ internal int InternalGetPosition()
             return _position;
         }
 
-        // PERF: Takes out Int32 as fast as possible
-        internal int InternalReadInt32()
+        // PERF: Takes out ReadOnlySpan as fast as possible
+        internal ReadOnlySpan<byte> InternalReadSpan(int count)
         {
             EnsureNotClosed();
 
-            int pos = (_position += 4); // use temp to avoid a race condition
-            if (pos > _length)
+            int origPos = _position;
+            int newPos = origPos + count;
```

Comment:
We should assert that count is > 0 && <= 16.  If count was sufficiently large, this could overflow.

Reply:
Fixed. but as I wrote above, I would like to make it public and made this a real check and not assert. waiting for your feedback.

Reply:
> Fixed. but as I wrote above, I would like to make it public and made this a real check and not assert. waiting for your feedback.

Waiting for my feedback on making it public?  If you want to expose additional APIs, you should open an issue for them per https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md.